### PR TITLE
Add generateTestHeaderStringAsync function to Webhooks.ts

### PR DESCRIPTION
**Pull request by zelief**
Currently, `generateTestHeaderString` throws a `CryptoProviderOnlySupportsAsyncError` when executed in Bun and other non-Node environments. This presents a limitation for users attempting to utilize this function outside of Node.js.

This PR addresses this issue by adding `generateTestHeaderStringAsync`. With this, creating a test header string in non-Node environments becomes more seamless and straightforward.

### CODE REVIEW

File: test/Webhook.spec.ts [hunk 1]
```diff
@@ -41,6 +41,30 @@ function createWebhooksTestSuite(stripe) {
...
+    it('should correctly construct a webhook header', async () => {
+      const header = await stripe.webhooks.generateTestHeaderStringAsync({
+        payload: EVENT_PAYLOAD_STRING,
+        secret: SECRET,
+      });
+
+      expect(header).to.not.be.undefined;
+      expect(header.split(',')).to.have.lengthOf(2);
+      expect(header).to.equal(
+        stripe.webhooks.generateTestHeaderString({
```
Comment (zelief) [inline test/Webhook.spec.ts:60]: Make sure the generated string is the same as `generateTestHeaderString`

**Comment by ramya-stripe** [comment]
Hey @zelief 

There is some duplicated code between `generateTestHeaderString` and `generateTestHeaderStringAsync`. Can you consider sharing code between the two to ensure they dont diverge in the future?

**Review by helenye-stripe — DISMISSED** [review]
See above comment.

**Comment by zelief** [comment]
> Hey @zelief
> 
> There is some duplicated code between `generateTestHeaderString` and `generateTestHeaderStringAsync`. Can you consider sharing code between the two to ensure they dont diverge in the future?

Sure @ramya-stripe, I will update it. Thanks for the review!

**Comment by zelief** [comment]
Updated @ramya-stripe, kindly check.

**Review by xavdid-stripe — APPROVED** [review]
Looks great, thank you!

**Pull request merged**